perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_attribute_def_virt_voPersonExternalAffiliation.java

```diff
@@ -125,8 +126,18 @@ public Attribute getAttributeValue(PerunSessionImpl sess, User user, AttributeDe
 		valuesWithoutDuplicities.addAll(getAffiliationsManuallyAssigned(sess, user));
 		valuesWithoutDuplicities.addAll(getAffiliationsFromGroups(sess, user));
 
+		// remove duplicities, by accepting only the first occurrence of each value (other occurences, case-insensitive, will be removed)
+		Set<String> valuesWithoutDuplicitiesCaseInsensitive = new HashSet<>();
```

Comment:
Nitpick, but this can be an ArrayList right away.

Why was this change necessary anyway?

Reply:
The Set performs better; it does not matter what we use.

The lower-case solution does not work because we shouldn't modify the values; we should group them case-insensitively.